Assets/Scripts/Models/Character/CharacterNameManager.cs

```diff
@@ -49,16 +42,20 @@ public static void LoadNames(string[] nameStrings)
     /// <returns>A randomly chosen name.</returns>
     public static string GetNewName()
     {
-        string name = unusedNames.Dequeue();
-        usedNames.Enqueue(name);
+        // If character names doesn't exist then just return null
+        if (characterNames.Length == 0)
+        {
+            return null;
+        }
 
-        // We run out of used names, lets start using the used ones
-        if (unusedNames.Count == 0)
+        // Re-Loop Pointer
+        if (ptr > 0 && ptr >= characterNames.Length)
```

Comment:
Its > 0 so it won't set 0 to 0, but honestly doesn't need to be there (since the array can't be length 0 so it'll never trigger as 0), so I'll remove it :D